app/views/homepage/_list_item.haml

```diff
@@ -32,6 +32,22 @@
             %a.home-share-type-link{:href => search_path(:transaction_type => shape_name_map[listing.listing_shape_id], :view => :list)}
               = icon_tag(listing.icon_name, ["icon-fix"])
               = shape_name(listing)
+      .home-list-price-mobile{:class => (listing.listing_images.size > 0 ? "home-list-price-mobile-with-listing-image" : "home-list-price-mobile-without-listing-image")}
```

Comment:
No change in content, just moved mobile price block `.home-list-price-mobile` inside the div containing listing title. The price block used to be absolutely positioned, which caused overlap. Now it's in normal page layout flow.